Sources/VLCRendererDiscoverer.m

```diff
@@ -153,16 +154,37 @@ - (void)dealloc
     return [list copy];
 }
 
+- (VLCRendererItem *)discoveredItemsContainsItem:(libvlc_renderer_item_t *)item
+{
+    for (VLCRendererItem *tmpItem in _items) {
+        if (tmpItem.rendererItem == item) {
+            return tmpItem;
+        }
+    }
+    return nil;
```

Comment:
what you could do is to have a method called addRendererItemIfNeeded

-(void) addRendererItemIfNeeded:(libvlc_renderer_item_t *)item {
  for (VLCRendererItem *tmpItem in _items) {
         if (tmpItem.rendererItem == item) {
             return;
         }
     }
   VLCRenderItem *tmpItem =  [[VLCRendererItem alloc] initWithCItem: pointerValue];
   [_items addObject: tmpItem];
   [_delegate rendererDiscovererItemAdded:self item:tmp];
}